### PR TITLE
[MRG+1] DOC: Updating docstring to numpy doc format for errorbar

### PR DESCRIPTION
**Pull request by Seraf69**
Hi matplotlib contributor,

we updated the docstring of errorbar to the numpy doc format. @NelleV 

Thanks for your patience,
Benedikt and Seraphin

### CODE REVIEW

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -2691,78 +2691,84 @@ def errorbar(self, x, y, yerr=None, xerr=None,
...
 
-          *errorevery*: positive integer
+        errorevery: positive integer, optional, default:1
             subsamples the errorbars. e.g., if errorevery=5, errorbars for
             every 5-th datapoint will be plotted. The data plot itself still
             shows all data points.
 
+        Returns
+        -------
+        plotline: :class:`~matplotlib.lines.Line2D` instance
```
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: For numpydoc to render properly, you need to add a space after plotline and before ":".

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -2691,78 +2691,84 @@ def errorbar(self, x, y, yerr=None, xerr=None,
...
+            x, y plot markers and/or line
+        caplines: 
+            list of error bar cap :class:`~matplotlib.lines.Line2D` instances
```
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: same

File: lib/matplotlib/axes/_axes.py [hunk 3]
```diff
@@ -2691,78 +2691,84 @@ def errorbar(self, x, y, yerr=None, xerr=None,
...
+        barlinecols:
```
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: same.

**Comment by NelleV** [comment]
Hello,

Thanks for the patch!
The matplotlib code follows the pep8 recommandations. The patch breaks a couple of these recommandations, mostly whitespace issues. Here is the traceback:

/home/travis/build/matplotlib/matplotlib/lib/matplotlib/axes/_axes.py:2705:19: W291 trailing whitespace
/home/travis/build/matplotlib/matplotlib/lib/matplotlib/axes/_axes.py:2707:1: W293 blank line contains whitespace
/home/travis/build/matplotlib/matplotlib/lib/matplotlib/axes/_axes.py:2708:80: E501 line too long (91 > 79 characters)
/home/travis/build/matplotlib/matplotlib/lib/matplotlib/axes/_axes.py:2764:18: W291 trailing whitespace
/home/travis/build/matplotlib/matplotlib/lib/matplotlib/axes/_axes.py:2766:21: W291 trailing whitespace
/home/travis/build/matplotlib/matplotlib/lib/matplotlib/axes/_axes.py:2767:80: E501 line too long (81 > 79 characters)
/home/travis/build/matplotlib/matplotlib/lib/matplotlib/axes/_axes.py:2792:1: W293 blank line contains whitespace

File: lib/matplotlib/axes/_axes.py [hunk 4]
```diff
@@ -2691,78 +2691,84 @@ def errorbar(self, x, y, yerr=None, xerr=None,
...
+        Horizontal errorbars are plotted if xerr is not None.
 
-        Optional keyword arguments:
+        x, y, xerr, and yerr can all be scalars, which plots a
+        single error bar at x, y.
 
-          *xerr*/*yerr*: [ scalar | N, Nx1, or 2xN array-like ]
+        Parameters
+        ----------
+        x : scalar
```
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: There is whitespace at the end of this line. Can you please remove it?

File: lib/matplotlib/axes/_axes.py [hunk 5]
```diff
@@ -2691,78 +2691,84 @@ def errorbar(self, x, y, yerr=None, xerr=None,
...
-        Optional keyword arguments:
+        x, y, xerr, and yerr can all be scalars, which plots a
+        single error bar at x, y.
 
-          *xerr*/*yerr*: [ scalar | N, Nx1, or 2xN array-like ]
+        Parameters
+        ----------
+        x : scalar            
+        y : scalar
+
```
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: There is whitespace on this line. Can you please remove it?

File: lib/matplotlib/axes/_axes.py [hunk 6]
```diff
@@ -2691,78 +2691,84 @@ def errorbar(self, x, y, yerr=None, xerr=None,
...
+        x, y, xerr, and yerr can all be scalars, which plots a
+        single error bar at x, y.
 
-          *xerr*/*yerr*: [ scalar | N, Nx1, or 2xN array-like ]
+        Parameters
+        ----------
+        x : scalar            
+        y : scalar
+            
+        xerr/yerr : scalar or array-like, shape(n,1) or shape(2,n), optional, default: None
```
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: This line is too long for our code requirements (max 79 char). Can you remove the default and add it in the text below?

File: lib/matplotlib/axes/_axes.py [hunk 7]
```diff
@@ -2691,78 +2691,84 @@ def errorbar(self, x, y, yerr=None, xerr=None,
...
+        errorevery: positive integer, optional, default:1
             subsamples the errorbars. e.g., if errorevery=5, errorbars for
             every 5-th datapoint will be plotted. The data plot itself still
             shows all data points.
 
+        Returns
+        -------
+        plotline: :class:`~matplotlib.lines.Line2D` instance
+            x, y plot markers and/or line
+        caplines:
```
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: There seems to be whitespace at the end of this line. Can you remove it?

File: lib/matplotlib/axes/_axes.py (hunk 3 above)
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: There is also whitespace at the end of this line :)

File: lib/matplotlib/axes/_axes.py [hunk 8]
```diff
@@ -2771,29 +2777,19 @@ def errorbar(self, x, y, yerr=None, xerr=None,
...
-                :class:`~matplotlib.collections.LineCollection` instances for
-                the horizontal and vertical error ranges.
-
-        **Example:**
 
+        Example
+        -------
         .. plot:: mpl_examples/statistics/errorbar_demo.py
-
+
```
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: And there is whitespace as well on this line!

I hope I got all of the whitespace issues.

**Comment by NelleV** [comment]
Apart from my comments, great job!

**Comment by NelleV** [comment]
There is also a slight problem of rendering: I am not sure what the problem is, but I'll look again once you update the PR with the changes. These might actually fix the rendering problem.

File: lib/matplotlib/axes/_axes.py [hunk 9]
```diff
@@ -2691,78 +2691,84 @@ def errorbar(self, x, y, yerr=None, xerr=None,
...
-          errorbar(x, y, yerr=None, xerr=None,
-                   fmt='', ecolor=None, elinewidth=None, capsize=None,
-                   barsabove=False, lolims=False, uplims=False,
-                   xlolims=False, xuplims=False, errorevery=1,
-                   capthick=None)
-
-        Plot *x* versus *y* with error deltas in *yerr* and *xerr*.
-        Vertical errorbars are plotted if *yerr* is not *None*.
-        Horizontal errorbars are plotted if *xerr* is not *None*.
+        """Plot an errorbar graph.
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: This should remain on a separate line; I think this is causing the rendering issue @NelleV noted.

File: lib/matplotlib/axes/_axes.py [hunk 10]
```diff
@@ -2691,78 +2691,84 @@ def errorbar(self, x, y, yerr=None, xerr=None,
...
 
-        *x*, *y*, *xerr*, and *yerr* can all be scalars, which plots a
-        single error bar at *x*, *y*.
+        Plot x versus y with error deltas in yerr and xerr.
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: I'm not sure that the arguments should not have the distinct formatting that they originally had.

**Comment by QuLogic** [comment]
It appears you have not [set the config for your git client](https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup#Your-Identity). All your commits are attributed to "Your Name <you@example.com>", which is probably not what you want.

File: lib/matplotlib/axes/_axes.py (hunk 10 above)
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: I much prefer the way it is now, and I think we've had this discussion about 3 years ago and agreed to remove the *. This do not bring much in the sphinx rendering, and clutter the python docstrings.

**Comment by daurer** [comment]
We tried to address all problems listed.

File: lib/matplotlib/axes/_axes.py [hunk 11]
```diff
@@ -2694,75 +2694,82 @@ def errorbar(self, x, y, yerr=None, xerr=None,
...
-        Optional keyword arguments:
+        x, y, xerr, and yerr can all be scalars, which plots a
+        single error bar at x, y.
 
-          *xerr*/*yerr*: [ scalar | N, Nx1, or 2xN array-like ]
+        Parameters
+        ----------
+        x : scalar
+        y : scalar
+
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: There are still spaces here.

File: lib/matplotlib/axes/_axes.py [hunk 12]
```diff
@@ -2694,106 +2694,101 @@ def errorbar(self, x, y, yerr=None, xerr=None,
...
-            *caplines*: list of error bar cap
-                :class:`~matplotlib.lines.Line2D` instances
-            *barlinecols*: list of
-                :class:`~matplotlib.collections.LineCollection` instances for
-                the horizontal and vertical error ranges.
-
-        **Example:**
 
+        Examples
+        -------
```
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: You have a tiny problem here: there is a dash missing to underline the whole section.

**Comment by NelleV** [comment]
This looks good! Thanks for the contribution.
Here is how it looks:

errorbar

**Comment by tacaswell** [comment]
backported to v2.x as 55d7a4f 

@Seraf69 Thanks!

**Pull request merged**